### PR TITLE
Included a regex replace in lists.getById() in order to be able to use e.g. _spPageContextInfo.pageListId directly

### PR DESCRIPTION
**Pull request by olemp**
| Q               | A
| --------------- | ---
| Bug fix?        | not sure :)
| New feature?    | yes
| New sample?      | no
| Related issues?  | fixes #X, partially #Y, mentioned in #Z

#### What's in this Pull Request?

Included a regex replace in lists.getById() in order to be able to use e.g. spPageContextInfo.pageListId directly. {} are replaced from the string.

Included a test.

### CODE REVIEW

File: src/sharepoint/rest/lists.ts [hunk 1]
```diff
@@ -41,7 +41,7 @@ export class Lists extends QueryableCollection {
      */
     public getById(id: string): List {
         let list = new List(this);
-        list.concat(`(guid'${id}')`);
+        list.concat(`(guid'${id.replace(/[^a-zA-Z0-9_-]/g,"")}')`);
```
Comment (patrick-rodgers) [inline src/sharepoint/rest/lists.ts]: Does it make sense to limit the regex to only supported characters (a-f)? Or to replace the "{" & "}" directly?

**Comment by olemp** [comment]
Not sure. Not a regex expert. Just replacing the {} might be sufficient!

**Comment by olemp** [comment]
@patrick-rodgers If we skip using guid'' we can use list guids with and without {}.

So 

` list.concat(`(${id})`);`

**Comment by patrick-rodgers** [comment]
Ok, that works for me. There may be a few other places where this is an issue.

**Comment by olemp** [comment]
@patrick-rodgers Fixed in https://github.com/OfficeDev/PnP-JS-Core/pull/108/commits/d8d47f204305e1ccf01285d645363353d447ad3a

File: tests/sharepoint/rest/lists.test.ts [hunk 2]
```diff
@@ -31,6 +31,12 @@ describe("Lists", () => {
             expect(list.toUrl()).to.equal("_api/web/lists(guid'4FC65058-FDDE-4FAD-AB21-2E881E1CF527')");
         });
     });
+    describe("getById with {}", () => {
+        it("Should return _api/web/lists('4FC65058-FDDE-4FAD-AB21-2E881E1CF527')", () => {
+            let list = lists.getById("{4FC65058-FDDE-4FAD-AB21-2E881E1CF527}");
+            expect(list.toUrl()).to.equal("_api/web/lists(guid'4FC65058-FDDE-4FAD-AB21-2E881E1CF527')");
```
Comment (patrick-rodgers) [inline tests/sharepoint/rest/lists.test.ts]: not sure this test passes now?

**Comment by patrick-rodgers** [comment]
Can you take a look at the test, not sure it passes as-is?

**Comment by olemp** [comment]
@patrick-rodgers Sorry, forgot about that part.. Commited https://github.com/OfficeDev/PnP-JS-Core/pull/108/commits/deffe36f06325a5d11989e2955d396244e93a1ab.

There's a change in the gulp task _test_ I see. How do I see the individual test results now? I can just see a summary.

**Comment by jansenbe** [comment]
Hi Ole,

Thx for this PR, we'll get it processed soon.

**Comment by patrick-rodgers** [comment]
If a test fails it shows up, otherwise just the summary. With so many tests listing them seemed like overkill - but if a test fails it is still shown with the error and expected vs actual value. I also added back the ' chars around the id after I tested this. Otherwise works well with both {...} and with out. Thanks Ole.

**Pull request merged**